x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -535,14 +536,14 @@ protected List<BucketExtractor> initBucketExtractors(SearchResponse response) {
             List<QueryContainer.FieldInfo> refs = query.fields();
 
             List<BucketExtractor> exts = new ArrayList<>(refs.size());
-            ConstantExtractor totalCount = new ConstantExtractor(response.getHits().getTotalHits().value);
+            Holder<ConstantExtractor> totalCountHolder = new Holder<>();
```

Comment:
Can you, please, explain the Holder usage here?

Reply:
The problem here is `response.getHits().getTotalHits().value`, that cannot be executed in case of `track_total_hits=false` (it would throw an error).
Using the holder I just postpone its execution until it's really needed, preserving the original logic (ie. instantiating only one `ConstantExtractor` and using it multiple times if needed). Though, it's really unlikely that we need more than one `ConstantExtractor`, and even in that case, having multiple instances should not be a problem.
So IMHO it's just a matter of preference, I can leave it as it is or I can remove the holder completely and instantiate a new `ConstantExtractor` every time it's needed.

Reply:
About EQL, I didn't check it yet. Since the two codebases do not seem to overlap on this fix, I'll provide a separate PR if relevant.